RubberduckTests/Grammar/SelectionExtensionsTests.cs

```diff
@@ -0,0 +1,494 @@
+﻿using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
+using static Rubberduck.Parsing.Grammar.VBAParser;
+using Rubberduck.VBEditor;
+using RubberduckTests.Mocks;
+using Antlr4.Runtime.Misc;
+using System.Collections.Generic;
+
+namespace RubberduckTests.Grammar
+{
+    [TestClass]
+    public class SelectionExtensionsTests
+    {
+        public class CollectorVBAParserBaseVisitor<Result> : VBAParserBaseVisitor<IEnumerable<Result>>
+        {
+            protected override IEnumerable<Result> DefaultResult => new List<Result>();
```

Comment:
I have to admit that a constant initializer would probably more efficient, i.e. replacing `=>` with `=`. That was a little oversight when I posted it in chat.

Reply:
Actually if I remove the `>` to make it constant initializer, compiler complains about it being not a valid override...